code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -382,9 +383,21 @@ private String getInstanceName(Instance instance) {
 
     @SuppressWarnings("unchecked")
     private Map<String, Object> getLaunchConfigDataAsMap(Service service) {
-        return (Map<String, Object>) DataAccessor.fields(service)
+        Map<String, Object> launchConfig = (Map<String, Object>) DataAccessor.fields(service)
                 .withKey(ServiceDiscoveryConstants.FIELD_LAUNCH_CONFIG).withDefault(Collections.EMPTY_MAP)
                 .as(Map.class);
+        Map<String, Object> data = new HashMap<>();
+        data.putAll(launchConfig);
+
+        Object labels = data.get(ServiceDiscoveryConfigItem.LABELS.getCattleName());
```

Comment:
This deep copy of labels is because when we're processing the labels for each deployment unit, we were inadvertently modifying the labels in the service.  So with each subsequent container getting deployed by the service for that particular service, it was getting labels from the previous container.  Hence, causing all the containers to get deployed to the same host.